### PR TITLE
feat: Uniformly distribute nodes across zones when zone is not specified

### PR DESCRIPTION
**Pull request by samuelstolicny**
## Summary
- Implements uniform distribution of nodes across availability zones when zone is not specified in the InputManifest
- Adds zone discovery data sources for all cloud providers (AWS, Azure, GCP, OCI)
- Uses round-robin distribution via modulo operator (`nodeIndex % zoneCount`)

### Provider-specific implementations:
- **AWS**: Creates per-node subnets with dynamic CIDR allocation (supports up to 256 nodes)
- **Azure**: Uses `azurerm_location` data source with null fallback for regions without AZ support
- **GCP**: Uses `google_compute_zones` data source
- **OCI**: Uses `oci_identity_availability_domains` with tenancy OCID

Closes https://github.com/berops/claudie/issues/1629

### CODE REVIEW

File: templates/terraformer/aws/nodepool/node_networking.tpl [hunk 1]
```diff
@@ -27,11 +30,51 @@ resource "aws_subnet" "{{ $subnetResourceName }}" {
...
+{{- $newbits := 4 }}{{- /* Default: supports up to 16 nodes */}}
+{{- if gt $nodeCount 16 }}{{- $newbits = 5 }}{{- end }}{{- /* 32 nodes */}}
+{{- if gt $nodeCount 32 }}{{- $newbits = 6 }}{{- end }}{{- /* 64 nodes */}}
+{{- if gt $nodeCount 64 }}{{- $newbits = 7 }}{{- end }}{{- /* 128 nodes */}}
+{{- if gt $nodeCount 128 }}{{- $newbits = 8 }}{{- end }}{{- /* 256 nodes */}}
+
+    {{- range $nodeIndex, $node := $nodepool.Nodes }}
+
+        {{- $subnetResourceName        := printf "%s_%s_%s_subnet" $nodepool.Name $node.Name $resourceSuffix }}
+        {{- $subnetName                := printf "snt-%s-%s-%s-%s" $clusterHash $region $nodepool.Name $node.Name }}
```
Comment (Despire) [inline templates/terraformer/aws/nodepool/node_networking.tpl:53]: **Have you also tested this with the maximum name lengths ?**

nodepools has a limit of 14 chars and cluster has a limit of 28 chars,
and the nodename is \<nodepool-name\>-<two_digit_number>

Some providers have limits on naming, want to know if this passes without issues.

File: templates/terraformer/oci/networking/networking.tpl [hunk 2]
```diff
@@ -19,6 +19,15 @@ locals {
 {{- range $_, $region := .Data.Regions }}
 
 {{- $resourceSuffix := printf "%s_%s_%s" $region $specName $uniqueFingerPrint }}
+{{- $varCompartmentID  := printf "default_compartment_id_%s" $resourceSuffix }}
```
Comment (Despire) [inline templates/terraformer/oci/networking/networking.tpl]: Why add another `$varCompartmentID` ? There are two now in the file, Or was it ment to be moved to this place and the old one removed ?

**Review by Despire — COMMENTED** [review]
Looks good, left some comments.

File: templates/terraformer/oci/networking/networking.tpl (hunk 2 above)
Comment (samuelstolicny) [reply templates/terraformer/oci/networking/networking.tpl]: good catch! Must be a eftover from when I changed the base branch from master to 0.9.16. .... Removed

File: templates/terraformer/aws/nodepool/node_networking.tpl (hunk 1 above)
Comment (samuelstolicny) [reply templates/terraformer/aws/nodepool/node_networking.tpl:53]: I havent specifically tested with maximum name lengths. During my tests I passed without issues with:
provider-name: "aws-provider"
cluster-name: "aws-cluster"
nodepool-name: "control-plane" (13 chars), "cpu-workers"

AWS allows 256 chars for subnet Name tags, so there is plenty of headroom.

**Review by Despire — APPROVED** [review]
